src/Features/Core/Portable/Snippets/SnippetProviders/AbstractConsoleSnippetProvider.cs

```diff
@@ -73,30 +74,19 @@ private async Task<TextChange> GenerateSnippetTextChangeAsync(Document document,
             return new TextChange(TextSpan.FromBounds(position, position), expressionStatement.NormalizeWhitespace().ToFullString());
         }
 
-        protected override int? GetTargetCaretPosition(ISyntaxFactsService syntaxFacts, SyntaxNode caretTarget)
+        protected override int GetTargetCaretPosition(ISyntaxFactsService syntaxFacts, SyntaxNode caretTarget)
         {
-            var invocationExpression = caretTarget.DescendantNodes().Where(syntaxFacts.IsInvocationExpression).FirstOrDefault();
-            if (invocationExpression is null)
-            {
-                return null;
-            }
-
-            var argumentListNode = syntaxFacts.GetArgumentListOfInvocationExpression(invocationExpression);
-            if (argumentListNode is null)
-            {
-                return null;
-            }
-
-            syntaxFacts.GetPartsOfArgumentList(argumentListNode, out var openParenToken, out _, out _);
-            return openParenToken.Span.End;
+            var openParenToken = GetOpenParenToken(caretTarget, syntaxFacts);
+            Contract.ThrowIfNull(openParenToken);
```

Comment:
why are we contract-throwing here?  how do you know this must insert an open paren?  (consdier VB, which would not have an open paren here).

Reply:
(Actually, the VB comment applies to if-statements, not console.writeline) (though in VB the parens are optinoal there too :))

Reply:
I'm creating the syntax, so I can ensure there are parentheses created. Not sure what you mean by it applying to if-statements, I don't get any open parentheses tokens in the if statement provider.